Tests/kaas/k8s-default-storage-class/k8s-default-storage-class-check.py

```diff
@@ -213,55 +328,49 @@ def main(argv):
             return 0
         if opt[0] == "-k" or opt[0] == "--kubeconfig":
             kubeconfig = opt[1]
-        else:
-            print_usage(kubeconfig)
-            return 2
-
-    # Setup kubernetes client
-    try:
-        logger.debug("setup_k8s_client(kubeconfig)")
-        k8s_core_api, k8s_storage_api = setup_k8s_client(kubeconfig)
-    except Exception as exception_message:
-        logger.info(f"{exception_message}")
-        return_message = f"{exception_message}"
-        return_code = 1
-
-    # Check if default storage class is defined (MENTETORY)
-    try:
-        logger.info("check_default_storageclass()")
-        default_class_name = check_default_storageclass(k8s_storage_api)
-    except SCSTestException as test_exception:
-        logger.info(f"{test_exception}")
-        return_message = f"{test_exception}"
-        return_code = test_exception.return_code
-    except Exception as exception_message:
-        logger.info(f"{exception_message}")
-        return_message = f"{exception_message}"
-        return_code = 1
-
-    # Check if default_persistent volume has ReadWriteOnce defined (MENTETORY)
-    try:
-        logger.info("check_default_persistentvolume_readwriteonce()")
-        return_code = check_default_persistentvolumeclaim_readwriteonce(
-            k8s_core_api, default_class_name
+        if opt[0] == "-d" or opt[0] == "--debug":
+            logging.getLogger().setLevel(logging.DEBUG)
+    if not kubeconfig:
+        logger.critical("You need to pass --kubeconfig=CLOUD.")
+        return 2
+
+    with TestEnvironment(kubeconfig) as env:
+        # Check if default storage class is defined (MANDATORY)
+        k8s_core_api = env.k8s_core_api
+        logger.debug("check_default_storageclass()")
+        try:
+            default_class_name = check_default_storageclass(env.k8s_storage_api)
+        except SCSTestException:
+            raise
+        except Exception:
+            env.return_code = 1
+            logger.debug("check_default_storageclass() failed")
+            return env.return_code
+        env.cleanup = True
+        try:
+            env.return_code = create_pvc_pod(k8s_core_api, default_class_name)
+        except ApiException as api_exception:
+            if api_exception.status == 409:
+                logger.info(
+                    "conflicting resources, "
+                    "try to clean up left overs, then start again"
+                )
+                env.return_code = 3
+                env.return_message = "(409) conflicting resources"
+                return
```

Comment:
I think the `return` is in error here.

Reply:
no it needs the return to get to the __exit__ where the error is handled and printed out (however, this whole exception might be unnedeeded as the conflicted resources is already dealed with in the prepare part )